---
 src/Makefile.am |    6 ++++++
 1 file changed, 6 insertions(+)

Index: xserver-xorg-video-ati/src/Makefile.am
===================================================================
--- xserver-xorg-video-ati.orig/src/Makefile.am	2007-08-23 08:14:37.000000000 +0200
+++ xserver-xorg-video-ati/src/Makefile.am	2007-08-23 08:25:14.000000000 +0200
@@ -187,3 +187,9 @@
 	theatre.h \
 	theatre_reg.h \
 	atipciids.h
+
+pcidatadir = $(datadir)/xserver-xorg/pci
+pcidata_DATA = ati.ids
+
+ati.ids: atipciids.h
+	awk '/^#define.*PCI_CHIP/ {print $$3}' ${srcdir}/atipciids.h | sed -e 's/0x/1002/' > ati.ids
