Index: xserver-xorg-video-ati/man/radeon.man
===================================================================
--- xserver-xorg-video-ati.orig/man/radeon.man	2007-05-21 00:59:25.000000000 +0200
+++ xserver-xorg-video-ati/man/radeon.man	2007-05-21 00:59:29.000000000 +0200
@@ -644,7 +644,7 @@
 Alan Hourihane           \fIalanh@fairlite.demon.co.uk\fP
 Marc Aurele La France    \fItsi@xfree86.org\fP
 Benjamin Herrenschmidt   \fIbenh@kernel.crashing.org\fP
-Michel Dänzer            \fImichel@tungstengraphics.com\fP
+Michel D\(:anzer            \fImichel@tungstengraphics.com\fP
 Alex Deucher             \fIalexdeucher@gmail.com\fP
 Bogdan D.                \fIbogdand@users.sourceforge.net\fP
 Eric Anholt              \fIeric@anholt.net\fP
