Index: xserver-xorg-video-ati/src/radeon_kms.c
===================================================================
--- xserver-xorg-video-ati.orig/src/radeon_kms.c	2011-01-31 19:05:30.008994413 +1100
+++ xserver-xorg-video-ati/src/radeon_kms.c	2011-01-31 19:18:22.550127483 +1100
@@ -826,6 +826,11 @@
     int ret;
 
     pScrn->fbOffset = 0;
+#if GET_ABI_MAJOR(ABI_VIDEODRV_VERSION) >= 9
+    pScreen->canDoBGNoneRoot = TRUE;
+#else
+    pScrn->canDoBGNoneRoot = TRUE;
+#endif
 
     miClearVisualTypes();
     if (!miSetVisualTypes(pScrn->depth,
