diff -Nurp patched//src/radeon_kms.c working//src/radeon_kms.c
--- patched//src/radeon_kms.c	2011-01-27 19:20:35.454881380 -0800
+++ working//src/radeon_kms.c	2011-01-27 19:54:54.411057080 -0800
@@ -840,6 +840,11 @@ Bool RADEONScreenInit_KMS(int scrnIndex,
     int ret;
 
     pScrn->fbOffset = 0;
+#if GET_ABI_MAJOR(ABI_VIDEODRV_VERSION) >= 9
+    pScrn->pScreen->canDoBGNoneRoot = TRUE
+#else
+    pScrn->canDoBGNoneRoot = TRUE;
+#endif
 
     miClearVisualTypes();
     if (!miSetVisualTypes(pScrn->depth,
